src/pymoc/modules/module_wrapper.py

```diff
@@ -89,7 +89,11 @@ def timestep_basin(self, dt=None):
     for neighbor in self.left_neighbors:
       wA -= neighbor.psi[-1] * 1e6
 
-    module.timestep(wA=wA, dt=dt)
+    if isinstance(module, SO_ML):
```

Comment:
We need to think about what to do when there are multiple basins and an interactive SO_ML. The model hasn't yet been used in such a configuration and it's not totally obvious to me what's the best way to go about this. I think the derivation for the basin exchange formally works only is the surface buoyancy in the SO is full zonally symmetric (i.e. the same in both SO sectors), which means there should never be more than one SO_ML sector. However, if there are two basins, there are two Psi_SO couplers and the SO_ML module should be using the total overturning (where the stream functions in the two sectors are added together at fixed b). We'll need to think about how to accomplish this, and it may require some modifications to the SO_ML module. Perhaps it would be worth to chat about this on the phone?

Reply:
Agreed, this is just about the last major change to address for this update. I should be available to chat sometime after this week